developer/src/kmc/test/test-infrastructureMessages.ts

```diff
@@ -36,31 +40,110 @@ describe('InfrastructureMessages', function () {
-    await testForMessage(this, ['invalid-keyboards', 'error_file_type_not_recognized.xxx'], CompilerMessages.ERROR_FileTypeNotRecognized);
+    const ncb = new NodeCompilerCallbacks({logLevel: 'silent'});
+    await unitTestEndpoints.build(makePathToFixture('invalid-keyboards', 'error_file_type_not_recognized.xxx'), '', ncb, {});
+    assert.isTrue(ncb.hasMessage(InfrastructureMessages.ERROR_FileTypeNotRecognized),
+      `ERROR_FileTypeNotRecognized not generated, instead got: `+JSON.stringify(ncb.messages,null,2));
   });
 
   // ERROR_OutFileNotValidForProjects
 
   it('should generate ERROR_OutFileNotValidForProjects if an output file is specified for a project build', async function() {
-    await testForMessage(this, ['invalid-keyboards', 'error_out_file_not_valid_for_projects.kpj'], CompilerMessages.ERROR_OutFileNotValidForProjects);
+    const ncb = new NodeCompilerCallbacks({logLevel: 'silent'});
+    const projectPath = makePathToFixture('kpj-2.0/khmer_angkor', 'khmer_angkor.kpj');
+    const outFilePath = makePathToFixture('kpj-2.0/khmer_angkor', 'khmer_angkor.kmx');
+    const options: CompilerOptions = {...defaultCompilerOptions};
+    await unitTestEndpoints.build(projectPath, outFilePath, ncb, options);
+    assert.isTrue(ncb.hasMessage(InfrastructureMessages.ERROR_OutFileNotValidForProjects),
+      `ERROR_OutFileNotValidForProjects not generated, instead got: `+JSON.stringify(ncb.messages,null,2));
   });
 
   // ERROR_InvalidProjectFile
 
   it('should generate ERROR_InvalidProjectFile if a project file is invalid', async function() {
-    await testForMessage(this, ['invalid-keyboards', 'error_invalid_project_file.kpj'], CompilerMessages.ERROR_InvalidProjectFile);
+    const ncb = new NodeCompilerCallbacks({logLevel: 'silent'});
+    await unitTestEndpoints.build(makePathToFixture('invalid-projects', 'error_invalid_project_file.kpj'), '', ncb, {});
+    assert.isTrue(ncb.hasMessage(InfrastructureMessages.ERROR_InvalidProjectFile),
+      `ERROR_InvalidProjectFile not generated, instead got: `+JSON.stringify(ncb.messages,null,2));
+  });
+
+  // ERROR_NotAProjectFile
+
+  it('should generate ERROR_NotAProjectFile if a project file is not the correct type', async function() {
+    const ncb = new NodeCompilerCallbacks({logLevel: 'silent'});
+    const projectPath = makePathToFixture('invalid-projects', 'error_not_a_project_file.xxx')
+    loadProject(projectPath, ncb);
+    assert.isTrue(ncb.hasMessage(InfrastructureMessages.ERROR_NotAProjectFile),
+    `ERROR_NotAProjectFile not generated, instead got: `+JSON.stringify(ncb.messages,null,2));
+  });
+
+  // ERROR_UnknownFileFormat
+
+  it('should generate ERROR_UnknownFileFormat if an analyze osk-char-use mapping file is not the correct type', async function() {
+    const ncb = new NodeCompilerCallbacks({logLevel: 'silent'});
+    const options = {mappingFile: makePathToFixture('analyze', 'error_not_a_project_file.xxx')};
```

Comment:
Even though it's busywork, I'd prefer the fixture to be duplicated, named `error_unknown_file_format.xxx`, for consistency :grin:

Reply:
The fixture was already there, it was just that the test had an incorrect filename.